src/MgvOfferMaking.sol

```diff
@@ -346,53 +343,89 @@ contract MgvOfferMaking is MgvHasOffers {
       }
 
       // insertion
-      Leaf leaf = offerList.leafs[insertionTick.leafIndex()];
+      Leaf leaf = offerList.leafs[insertionTick.leafIndex()].clean();
       // if leaf was empty flip tick on at level0
       if (leaf.isEmpty()) {
         Field field;
         int insertionIndex = insertionTick.level0Index();
         int currentIndex = cachedLocalTick.level0Index();
         // Get insertion level0
         if (insertionIndex != currentIndex) {
-          field = offerList.level0[insertionIndex];
+          field = offerList.level0[insertionIndex].clean();
         } else {
           field = ofp.local.level0();
         }
 
         // Save current level0
         if (insertionIndex < currentIndex) {
-          offerList.level0[currentIndex] = ofp.local.level0();
+          Field localLeve0 = ofp.local.level0();
+          bool shouldSaveLevel0 = !localLeve0.isEmpty();
+          if (!shouldSaveLevel0) {
+            shouldSaveLevel0 = !offerList.level0[currentIndex].eq(DirtyFieldLib.CLEAN_EMPTY);
+          }
+          if (shouldSaveLevel0) {
+            offerList.level0[currentIndex] = localLeve0.dirty();
+          }
```

Comment:
Would an `else` be cheaper here?

```suggestion
          if (!shouldSaveLevel0) {
            shouldSaveLevel0 = !offerList.level0[currentIndex].eq(DirtyFieldLib.CLEAN_EMPTY);
          } else {
            offerList.level0[currentIndex] = localLeve0.dirty();
          }
```

Reply:
I don't think this results in the same logic. Suppose that `shouldSaveLevel0=false` and that `offerList.level[currentIndex].eq(DirtyFieldLib.CLEAN_EMPTY)=false`.
* As-is, the first `if` will update `shouldSaveLevel0` to `true`. Then the second `is` will update the `level0` mapping.
* With the suggestion, the first branch of the only `if` will be taken, and then the mapping will not be updated.

Reply:
doh! I don't know how I missed that 🤦‍♂️